src/walmart/WalmartClient.js

```diff
@@ -0,0 +1,196 @@
+const fetch = require('isomorphic-fetch');
+const Promise = require('bluebird');
+const fs = require('fs');
+
+/** Class for handling operations with Walmart's inventory */
+class WalmartClient {
+  // Public methods
+
+  /*
+  Instance Attributes
+  apiKey - The Walmart developer api key. Required for authenticating requests.
+  categoryIds - Ids Walmart uses to represent store product categories.
+  delayTime - Amount of time (ms) to delay in between requests. This can be useful when making
+              hundreds of requests since there is a limit to how many can run asynchronously at once.
+  */
+  constructor() {
+    this.apiKey = process.env.WALMART_API_KEY;
+    // Walmart category IDs will likely not change often so hard coding these in to avoid an extra api request.
+    // Can always call this.getCategoryIds() or this.writeCategoryIdsToFile() to update the list if we think they change.
+    this.categoryIds = [
```

Comment:
I suggest calling the function instead of hard-coding IDs. I can't justify the overhead of a single api request given the thought that we could miss out on a new category and the potential benefits or we could be running 100s of request for a category that has been removed but we still have in our list (think paginated calls).

Reply:
Removed the hard coding and now make an api request for the categories. https://github.com/sulaimonlasisi/clearance-app/pull/9/commits/a17ebd6f05edd6cc0294c64d944f3deeb8e39fdc and https://github.com/sulaimonlasisi/clearance-app/pull/9/commits/e61ebaffbf3b72f77a1ab3115e84652f0a527242